File: src/rag/image/search_image.py
def search_images(db, query_embedding, k=3):
    """
    Realiza búsqueda vectorial en la colección 'media'.
    Requiere un índice Atlas Search tipo vector llamado 'image_vector_index'.
    """
    pipeline = [
        {
            "$vectorSearch": {
                "index": "image_vector_index",
                "path": "image_embedding",
                "queryVector": query_embedding,
                "numCandidates": 200,
                "limit": k,
                "similarity": "cosine"
            }
        },
        {
            "$addFields": {
                "score": { "$meta": "vectorSearchScore" }
            }
        },
        {
            "$project": {
                "title": 1,
                "category": 1,
                "tags": 1,
                "caption": 1,
                "score": 1,
                "image_file_id": 1
            }
        }
    ]

    return list(db["media"].aggregate(pipeline))

File: src/rag/image/search_image.js
// src/rag/image/search_image.js
async function searchImages(db, queryEmbedding, k = 5) {
    return db.collection("media").aggregate([
        {
            $vectorSearch: {
                index: "media_embedding_index",
                path: "embedding",
                queryVector: queryEmbedding,
                numCandidates: 200,
                limit: k
            }
        },
        {
            $addFields: { score: { $meta: "vectorSearchScore" } }
        }
    ]).toArray();
}

module.exports = { searchImages };
